examples/pytorch/cluster_gcn/cluster_gcn.py

```diff
@@ -0,0 +1,247 @@
+        g.remove_edges_from(g.selfloop_edges())
+        g.add_edges_from(zip(g.nodes(), g.nodes()))
+        print("adding self-loop edges")
+    g = DGLGraph(g, readonly=True)
+
+    # set device for dataset tensors
+    if args.gpu < 0:
+        cuda = False
+    else:
+        cuda = True
+        torch.cuda.set_device(args.gpu)
+        features = features.cuda()
+        labels = labels.cuda()
+        train_mask = train_mask.cuda()
+        val_mask = val_mask.cuda()
+        test_mask = test_mask.cuda()
+
+    print(torch.cuda.get_device_name(0))
+
+    g.ndata['features'] = features
+    g.ndata['labels'] = labels
+    g.ndata['train_mask'] = train_mask
+    print('labels shape:', labels.shape)
+
+    cluster_iterator = ClusterIter(
+        args.dataset, g, args.psize, args.batch_size, train_nid, use_pp=args.use_pp)
+
+    print("features shape, ", features.shape)
+
+    model_sel = {'GCN': GCNCluster, 'graphsage': GraphSAGE}
+    model_class = model_sel[args.model_type]
+    print('using model:', model_class)
+
+    model = model_class(in_feats,
+                        args.n_hidden,
+                        n_classes,
+                        args.n_layers,
+                        F.relu,
+                        args.dropout, args.use_pp)
+
+    if cuda:
+        model.cuda()
+
+    # logger and so on
+    log_dir = save_log_dir(args)
+    writer = SummaryWriter(log_dir)
+    logger = Logger(os.path.join(log_dir, 'loggings'))
+    logger.write(args)
+
+    # Loss function
+    if multitask:
+        print('Using multi-label loss')
+        loss_f = nn.BCEWithLogitsLoss()
+    else:
+        print('Using multi-class loss')
+        loss_f = nn.CrossEntropyLoss()
+
+    # use optimizer
+    optimizer = torch.optim.Adam(model.parameters(),
+                                 lr=args.lr,
+                                 weight_decay=args.weight_decay)
+
+    # initialize graph
+    dur = []
+
+    # set train_nids to cuda tensor
+    if cuda:
+        train_nid = torch.from_numpy(train_nid).cuda()
+    print("current memory after model before training",
+          torch.cuda.memory_allocated(device=train_nid.device) / 1024 / 1024)
+    start_time = time.time()
+    best_f1 = -1
+
+    for epoch in range(args.n_epochs):
+        for j, cluster in enumerate(cluster_iterator):
+            # sync with upper level training graph
+            cluster.copy_from_parent()
+            model.train()
+            # forward
+            pred = model(cluster)
+            batch_labels = cluster.ndata['labels']
+            batch_train_mask = cluster.ndata['train_mask']
+            loss = loss_f(pred[batch_train_mask],
+                          batch_labels[batch_train_mask])
+
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            if j % args.log_every == 0:
+                print(f"epoch:{epoch}/{args.n_epochs}, Iteration {j}/{len(cluster_iterator)}:training loss", loss.item())
```

Comment:
Seems I'am always getting "Iteration 0/50" when using PPI.

Reply:
Because I set the `log_every` to be a `100`, which is bigger than ppi's partition size per epoch. If `log_every` is set to be `10`, then every epoch it will print log 5 times. I think it could be useful if one wants to train on a large graph for a few epochs.

Reply:
I C. Then maybe add a comment there so people can check the comment and understand what's going on.